modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -1867,12 +1893,28 @@ private void resizeCellSize(T cell) {
         }
     }
 
-    private List<T> getCells() {
+    /**
+     * Returns the list of cells displayed in the current viewport.
+     * <p>
+     * The cells are ordered such that the first cell in this list is the first in the view, and the last cell is the
+     * last in the view. When pixel scrolling, the list is simply shifted and items drop off the beginning or the end,
+     * depending on the order of scrolling.
+     *
+     * @return the cells displayed in the current viewport
+     * @since 12
+     */
+    protected List<T> getCells() {
         return cells;
     }
 
-    // Returns last visible cell whose bounds are entirely within the viewport
-    T getLastVisibleCellWithinViewPort() {
+    /**
+     * Returns the last visible cell whose bounds are entirely within the viewport. When manually inserting rows, one
+     * may need to know which cell indices are visible in the viewport.
+     *
+     * @return last visible cell whose bounds are entirely within the viewport
+     * @since 12
+     */
+    protected T getLastVisibleCellWithinViewPort() {
```

Comment:
'viewport' is one word, so this should be renamed to `getLastVisibleCellWithinViewport`. Ditto for getFirst...